tools/redir-gen/main.go

```diff
@@ -0,0 +1,184 @@
+	allRepos := riSlice{}
+	client := github.NewClient(nil)
+	for _, org := range orgs {
+		opts := &github.RepositoryListByOrgOptions{
+			ListOptions: github.ListOptions{PerPage: 50},
+		}
+		for {
+			repos, resp, err := client.Repositories.ListByOrg(ctx, org, opts)
+			if err != nil {
+				return nil, err
+			}
+			for _, r := range repos {
+				if !allowedRepoRe.MatchString(*r.Name) {
+					log.Printf("Ignoring repo %s, matched by ignore %q", *r.Name, allowedRepoRe)
+					continue
+				}
+				if *r.Archived {
+					log.Print("Ignoring archived repo: ", *r.Name)
+					continue
+				}
+				allRepos = append(allRepos, repoInfo{org, *r.Name, *r.DefaultBranch})
+			}
+			if resp.NextPage == 0 {
+				break
+			}
+		}
+	}
+	sort.Sort(allRepos)
+	return allRepos, nil
+}
+
+const (
+	goHTML = `<html><head>
+    <meta name="go-import" content="knative.dev/{{.Repo}} git https://github.com/{{.Org}}/{{.Repo}}">
+    <meta name="go-source" content="knative.dev/{{.Repo}}     https://github.com/{{.Org}}/{{.Repo}} https://github.com/{{.Org}}/{{.Repo}}/tree/{{.DefaultBranch}}{/dir} https://github.com/{{.Org}}/{{.Repo}}/blob/{{.DefaultBranch}}{/dir}/{file}#L{line}">
+</head></html>
+`
+	redirText = `/{{.Repo}}/* go-get=1 /golang/{{.Repo}}.html 200
+`
+
+	autogenPrefix = `
+# This file is AUTO-GENERATED
+#
+# DO NOT EDIT!
+#
+# To regenerate, run:
+`
+)
+
+func appendRedirs(ris []repoInfo) error {
+	redirFilename := "golang/_redirects"
+	redirFile, err := os.OpenFile(redirFilename, os.O_RDWR|os.O_CREATE, 0755)
+	if err != nil {
+		return fmt.Errorf("unable to open %q: %w", redirFilename, err)
+	}
+	defer redirFile.Close()
+
+	redirs, err := ioutil.ReadAll(redirFile)
+	if err != nil {
+		return fmt.Errorf("unable to read %q: %w", redirFilename, err)
+	}
+
+	seekTo := int64(strings.Index(string(redirs), autogenPrefix))
+	if seekTo == -1 {
+		seekTo = int64(len(redirs))
+	}
+	if _, err := redirFile.Seek(seekTo, 0); err != nil {
+		return fmt.Errorf("unable to seek in %q: %w", redirFilename, err)
+	}
+	if redirFile.Truncate(seekTo) != nil {
+		return fmt.Errorf("unable to truncate %q: %w", redirFilename, err)
+	}
+
+	if _, err := redirFile.WriteString(autogenPrefix); err != nil {
+		return fmt.Errorf("unable to write to %q: %w", redirFilename, err)
+	}
+	if _, err := redirFile.WriteString(fmt.Sprintf("#   %s\n", strings.Join(os.Args, " "))); err != nil {
+		return fmt.Errorf("unable to write to %q: %w", redirFilename, err)
+	}
+	for _, ri := range ris {
+		if redirTemplate.Execute(redirFile, ri) != nil {
+			return fmt.Errorf("unable to write %s to %q: %w", ri.Repo, redirFilename, err)
+		}
+	}
+	return nil
+}
+
+var fileTemplate = template.Must(template.New("gohtml").Parse(goHTML))
+var redirTemplate = template.Must(template.New("redir").Parse(redirText))
+
+// createGoGetFile creates a static HTML file providing a knative.dev mapping
+// for the specified org and repo.
+func createGoGetFile(ri repoInfo) error {
+	filename := fmt.Sprintf("golang/%s.html", ri.Repo)
```

Comment:
nit
this fails if `golang` directory doesn't exist. I worked around create the directory first.
```
2021/06/15 16:09:23 Unable to create go mod file for knative-sandbox/actions-downstream-test: open golang/actions-downstream-test.html: no such file or directory
```

Reply:
How can it not exist? The directory is added in this commit

Reply:
humm interesting, let me double check

Reply:
I ran it from root of the repo and got some problem, then I ran it from the tools directory

Reply:
Ah I got this when I tried to run the script
```
❯ go run tools/redir-gen/main.go
tools/redir-gen/main.go:27:2: cannot find package "." in:
	/Users/csantana23/dev/knative/docs/vendor/github.com/google/go-github/v32/github
```

Reply:
How is the script suppose to be run?

Reply:
Then I did
```
❯ cd tools/redir-gen/
❯ go run main.go
```

Reply:
Instructions in the PR description :)

> To generate redirects again (e.g. if a new repo is added to one of the knative orgs):
> 
> Go to tools/redir-gen, do go install .
> Run redir-gen in root folder of docs.